xarray/backends/h5netcdf_.py

```diff
@@ -43,16 +43,33 @@
     from xarray.core.dataset import Dataset
     from xarray.core.datatree import DataTree
     from xarray.core.types import ReadBuffer
+    from xarray.namedarray._typing import (
+        _BasicIndexerKey,
+        _OuterIndexerKey,
+        _VectorizedIndexerKey,
+    )
 
 
 class H5NetCDFArrayWrapper(BaseNetCDF4Array):
+    indexing_support = indexing.IndexingSupport.OUTER_1VECTOR
+
     def get_array(self, needs_lock=True):
         ds = self.datastore._acquire(needs_lock)
         return ds.variables[self.variable_name]
 
-    def __getitem__(self, key):
-        return indexing.explicit_indexing_adapter(
-            key, self.shape, indexing.IndexingSupport.OUTER_1VECTOR, self._getitem
+    def _oindex_get(self, key: _OuterIndexerKey) -> Any:
+        return indexing.outer_indexing_adapter(
```

Comment:
now we have three different adapters . This kind of dispatching was happening in `explicit_indexing_adapter` in any case...